assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -1172,130 +1172,153 @@ public String buildMainPanelContext(VelocityPortlet portlet, Context context, Ru
             context.put("allowGradeSubmission", state.getAttribute(STATE_ALLOW_GRADE_SUBMISSION));
         }
 
-        if (MODE_LIST_ASSIGNMENTS.equals(mode)) {
-            // build the context for the student assignment view
-            template = build_list_assignments_context(portlet, context, data, state);
-        } else if (MODE_STUDENT_VIEW_ASSIGNMENT.equals(mode)) {
-            // the student view of assignment
-            template = build_student_view_assignment_context(portlet, context, data, state);
-        } else if (MODE_STUDENT_VIEW_GROUP_ERROR.equals(mode)) {
-            // disable auto-updates while leaving the list view
-            justDelivered(state);
-
-            // build the context for showing group submission error
-            template = build_student_view_group_error_context(portlet, context, data, state);
-        } else if (MODE_STUDENT_VIEW_SUBMISSION.equals(mode)) {
-            // disable auto-updates while leaving the list view
-            justDelivered(state);
-
-            // build the context for showing one assignment submission
-            template = build_student_view_submission_context(portlet, context, data, state);
-        } else if (MODE_STUDENT_VIEW_SUBMISSION_CONFIRMATION.equals(mode)) {
-            context.put("site", s);
-
-            // build the context for showing one assignment submission confirmation
-            template = build_student_view_submission_confirmation_context(portlet, context, data, state);
-        } else if (MODE_STUDENT_PREVIEW_SUBMISSION.equals(mode)) {
-            // build the context for showing one assignment submission
-            template = build_student_preview_submission_context(portlet, context, data, state);
-        } else if (MODE_STUDENT_VIEW_GRADE.equals(mode) || MODE_STUDENT_VIEW_GRADE_PRIVATE.equals(mode)) {
-            context.put("site", s);
-
-            // disable auto-updates while leaving the list view
-            justDelivered(state);
-
-            if (MODE_STUDENT_VIEW_GRADE_PRIVATE.equals(mode)) {
+        switch (mode) {
```

Comment:
I'm not sure if `mode` can be null, but we were previously protected from NPEs because the `.equals` on the String literal is null safe.

It might be worthwhile to add a null check here, something like: `if (StringUtils.isNotEmpty(mode)) switch (mode) { ... }`

Reply:
Addressed this before switch(mode)

Reply:
Yes, I see that now.